### PR TITLE
Some pages are broken due to request store in the db and builder not …

### PR DESCRIPTION
**Pull request by mariangemarcano**
…longer available

### CODE REVIEW

**Comment by campos-ddc** [comment]
Can we write a test for this?

**Comment by campos-ddc** [comment]
Otherwise, code looks OK

File: master/buildbot/status/buildrequest.py [hunk 1]
```diff
@@ -165,6 +170,10 @@ def asDict(self):
     def asDict_async(self, request=None):
         result = {}
 
+        builder = self.status.getBuilder(self.getBuilderName())
+        if not builder:
+            defer.returnValue(result)
+            return
```
Comment (opdude) [inline master/buildbot/status/buildrequest.py]: I'm not sure if there is "good" time where we have no builder associated but... if we did then wouldn't this return mean that we won't get some of the info like source stamps?

Comment (mariangemarcano) [inline master/buildbot/status/buildrequest.py]: I will double check, but buildrequest depends on the builder being present on the master so most of the process wont work, maybe we could figure out this earlier.

**Comment by mariangemarcano** [comment]
Will update the pr for vKatana2.1.x/staging branch instead

**Comment by mariangemarcano** [comment]
Will update the pr for vKatana2.1.x/staging branch instead

**Pull request closed without merging**